File: food-review/src/main/java/com/food/review/domain/Review.java
package com.food.review.domain;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import javax.validation.constraints.Min;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.PastOrPresent;
import javax.validation.constraints.Size;
import java.io.Serializable;
import java.time.LocalDateTime;
import java.util.UUID;

@Getter
@Setter(AccessLevel.PRIVATE)
@Builder(toBuilder = true)
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@NoArgsConstructor(access = AccessLevel.PRIVATE)
@ToString(exclude = {"id", "username"})
@EqualsAndHashCode
public class Review implements Serializable {

    private static final long serialVersionUID = 8682067039933748251L;

    private UUID id;

    @NotNull
    private UUID orderId;

    @NotBlank
    private String username;

    @NotBlank
    private String name;

    @NotBlank
    @Size(min = 10, max = 500)
    private String text;

    @NotNull
    @PastOrPresent
    private LocalDateTime createdAt;

}

File: food-review/src/main/kotlin/com/food/review/domain/Review.kt
package com.food.review.domain

import com.food.review.config.annotation.Default
import java.time.LocalDateTime
import java.util.UUID

class Review @Default constructor(
    val id: String?,
    val uuid: UUID,
    val status: ReviewStatus,
    val order: Order,
    val text: String?,
    val createdAt: LocalDateTime?,
    val updatedAt: LocalDateTime?) {

    constructor(status: ReviewStatus, order: Order) :
            this(null, UUID.randomUUID(), status, order, null, null, null)
}
